### DISCUSSION TITLE
Inquiry regaridng BIMP design and creation

### DISCUSSION THREAD

**Discussion by Mortal-Engine** [alessandrofrancesconi/gimp-plugin-bimp#329] [General]
@alessandrofrancesconi How did you develop this app? Were you previously familiar with the GIMP API and did that permit you to make a plugin? I'm curious as I see several plugins for blender as well and am wondering how to contribute. Thanks.

**Comment by alessandrofrancesconi** [comment]
Thanks for the question! This was my first GIMP plugin so I started with [this resource](https://developer.gimp.org/writing-a-plug-in/1/index.html) for a first overview on how to write the codebase. Please note, I already knew C programming language so I that "just" a matter of using the proper GIMP API functions to start writing something.

Talking about APIs, this was (and still it is) a bible to understand all the available functions to manipulate images with GIMP: https://developer.gimp.org/api/2.0/libgimp/index.html

I was not familiar with GIMP API already, I had to look around these resources, and Google in general on how to do many things. Another way to learn is to look at other plugins (written in C or Python, it depends on your programming knowledge and needs), to see how they manage to some features you want to implement, too.

Also, if you have a plugin in mind that needs a graphical user interface, you should consider using [GTK+](https://www.gtk.org/), as BIMP does.

"Contributing" means writing your own plugin from scratch, or maybe start adding more features and fixes to already developed programs (like the Blender's plugins you talk about). I hope I gave you the right inputs, let me know if you have other questions!

**Comment by Mortal-Engine** [comment]
Thank you. To be frank, I'm not even sure of where to start with even accessing the core GIMP files. Do you have any books or recommendations that might elucidate how to expose myself to certain application configurations so that I might make them more accessible?
